Assets/HoloToolkit/Utilities/Scripts/Performance/ShaderControl.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    public class ShaderControl : MonoBehaviour
+    {
+        [SerializeField]
+        private Material[] materialList = new Material[1];
+
+        private int currentShaderLevel = 0;
+        private Renderer objectRenderer;
+
+        private void Awake()
+        {
+            objectRenderer = GetComponent<Renderer>();
+            if (materialList.Length > 0 && materialList[0] == null)
+            {
+                materialList[0] = objectRenderer.material;
```

Comment:
This is a very bad documented memory leak in Unity.  We should also be destroying the `materialList` material instances we make here in OnDestory.